File: hw/vaca/u3/HWObserver/Observer/src/main/java/ec/edu/espe/observer/view/Investor.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.observer.view;

import ec.edu.espe.observer.model.Stock;

/**
 *
 * @author LABS-ESPE
 */
public class Investor implements IInvestor {
    private String name;

    public Investor(String name) {
        this.name = name;
    }

    @Override
    public void update(Stock stock, Object arg) {
        System.out.println("Notified observer " + name);
        if (arg instanceof Double) {
            System.out.println("The price of " + stock.getSymbol() + " changed to: " + arg);
        } else if (arg instanceof String) {
            System.out.println("The symbol of " + stock.getSymbol() + " changed to: " + arg);
        }
    }
}

File: ws/fuentes/U3/WS35OBSERVER/OBSERVERPY/ec/edu/espe/saleApp/model/investor.py
from model.stock import IInvestor, Stock

class Investor(IInvestor):
    def __init__(self, name: str):
        self._name = name
        self._observer_state = ""
        self._stock = None

    def update(self, stock: Stock, arg):
        print(f"Notified observer {self._name}")

        if isinstance(arg, str):
            print(f"The symbol of {stock.get_symbol()} changed to: {arg}")
        elif isinstance(arg, float):
            print(f"The price of {stock.get_symbol()} changed to: {arg}")
